pkg/controller/usercleanup/user_cleanup_controller_test.go

```diff
@@ -0,0 +1,132 @@
+package usercleanup
+
+import (
+	"context"
+	"github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/host-operator/pkg/apis"
+	"github.com/codeready-toolchain/host-operator/pkg/configuration"
+	"github.com/codeready-toolchain/host-operator/pkg/metrics"
+	test2 "github.com/codeready-toolchain/host-operator/test"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+func TestUserCleanup(t *testing.T) {
+	t.Run("test that user cleanup doesn't delete an active UserSignup", func(t *testing.T) {
+		userSignup := &v1alpha1.UserSignup{
+			ObjectMeta: test2.NewUserSignupObjectMeta("", "abigail.thompson@redhat.com"),
+			Spec: v1alpha1.UserSignupSpec{
+				UserID:      "User-98887",
+				Username:    "abigail.thompson@redhat.com",
+				Deactivated: false,
+			},
+			Status: v1alpha1.UserSignupStatus{
+				Conditions: []v1alpha1.Condition{
+					{
+						Type:   v1alpha1.UserSignupComplete,
+						Status: v1.ConditionTrue,
+					},
+					{
+						Type:   v1alpha1.UserSignupApproved,
+						Status: v1.ConditionTrue,
+						Reason: "ApprovedAutomatically",
+					},
+				},
+				CompliantUsername: "abigail-thompson",
+			},
+		}
+		userSignup.Labels["toolchain.dev.openshift.com/approved"] = "true"
+		userSignup.Labels[v1alpha1.UserSignupStateLabelKey] = "approved"
```

Comment:
Please, for the creation of the UserSignups use the functions from the `test` package: https://github.com/codeready-toolchain/host-operator/blob/master/test/usersignup.go

Reply:
I've looked at these functions.  It seems there's no simple way to create a `UserSignup` with the same spec values (including `Deactivated`), conditions and labels, or am I missing something?

Reply:
you can use the modifier functions for that https://github.com/codeready-toolchain/host-operator/blob/master/test/usersignup.go#L16-L56
and create new ones if needed. Usage of these functions makes the test shorter, cleaner, and it's easier to focus on what is important to be configured - at least from my point of view :wink: .

Reply:
No problem, I've replaced `UserSignup` creation with the functions and added new modifier functions where required.